### PR TITLE
Adds feature to generate CITATION.cff from citation info found in description

### PR DESCRIPTION
**Pull request by SimaoBolota-MetaCell**
Pull Request for the main Citation Logic
---

**Summary**

This utility works by looking into:

 - GitHub Repository metadata
 - README.md
 
and creating a CITATION.CFF, a plain text file with human- and machine-readable citation information for software and datasets.

For citations to be parsed from the README.md these need to have either the **APA**(American Psychological Association) style or **BibTex** style format.

The **CITATION.CFF** file naming needs to be as it is, otherwise GitHub won't recognize it as a citation file.

To create the CITATION.CFF for your plugin run

```
 $ napari-hub-cli create-cff-citation /tmp/example-plugin 
```
having first installed all the requirements for ```napari-hub-cli```

The format for the CITATION.CFF is the following:
- The first citation that appears references the software
- The preffered citation, the one used in GitHub, references an article/publisher, preferring an article citation when both exist
- when both article and publisher information exist, publisher information is written as a sub-reference

### CODE REVIEW

**Comment by neuromusic** [comment]
I'm getting a ModuleNotFoundError on a clean install:

``` bash
(nhcli) ➜  ~ napari-hub-cli create-cff-citation Code/cellfinder-napari
Traceback (most recent call last):
  File "/Users/justin.kiggins/opt/anaconda3/envs/nhcli/bin/napari-hub-cli", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/Users/justin.kiggins/Code/napari-hub-cli/napari_hub_cli/cli.py", line 119, in main
    args.func(args)
  File "/Users/justin.kiggins/Code/napari-hub-cli/napari_hub_cli/cli.py", line 68, in create_citation
    from create_citation.citation import cff_citation
ModuleNotFoundError: No module named 'create_citation'
```

File: napari_hub_cli/cli.py [hunk 1]
```diff
@@ -57,6 +57,23 @@ def check_missing(args):
...
 
+def create_citation(args):
+    """Creates the CFF citation for the plugin at args.plugin_path
+
+    Parameters
+    ----------
+    args : List[str]
+        List of command line arguments
+    """
+    from create_citation.citation import cff_citation
```
Comment (neuromusic) [inline napari_hub_cli/cli.py:68]:
```suggestion
    from napari_hub_cli.create_citation.citation import cff_citation
```

File: setup.cfg [hunk 2]
```diff
@@ -29,6 +29,12 @@ install_requires =
     pandas
     parsesetup
     PyYAML
+    GitPython
+    beautifulsoup4
+    HTMLParser
```
Comment (neuromusic) [inline setup.cfg:34]:
```suggestion
    HTMLParser
    html5lib
```

**Review by neuromusic — CHANGES_REQUESTED** [review]

**Comment by neuromusic** [comment]
I've added 2 suggested changes that let me install and run this locally.

However, I'm a bit confused about usage still. Following the README, I get a JSON printed to the command line:

``` sh
A̲P̲A̲ ̲C̲i̲t̲a̲t̲i̲o̲n̲ 


{'cff-version': '1.2.0', 'message': 'If you use this plugin, please cite it using these metadata', 'title': 'cellfinder-napari', 'authors': [{'family-names': 'Justin', 'given-names': 'Kiggins'}], 'url': 'https://github.com/neuromusic/cellfinder-napari', 'date-released': '2021-01-01', 'preferred-citation': {'type': 'journal', 'journal': ', cellfinder-napari, cellfinder and cellfinder-core were developed by Charly Rousseau and Adam Tyson in the Margrie Lab, based on previous work by Christian Niedworok, generously supported by the Sainsbury Wellcome Centre., , Visualising detected cells in the cellfinder napari plugin, Once you have installed napari.\nYou can install napari either through the napari plugin installation tool, or\ndirectly from PyPI with:, Full documentation can be\nfound here., This software is at a very early stage, and was written with our data in mind.\nOver time we hope to support other data types/formats. If you have any\nquestions or issues, please get in touch by\nemail,\non the forum or by\nraising an issue., cellfinder takes a stitched, but otherwise raw dataset with at least\ntwo channels:, \nRaw coronal serial two-photon mouse brain image showing labelled cells, Classical image analysis (e.g. filters, thresholding) is used to find\ncell-like objects (with false positives):, \nCandidate cells (including many artefacts), A deep-learning network (ResNet) is used to classify cell candidates as true\ncells or artefacts:, \nCassified cell candidates. Yellow - cells, Blue - artefacts, If you find this plugin useful, and use it in your research, please cite the preprint outlining the cell detection algorithm:, Tyson, A., Rousseau, C., Niedworok, C., Keshavarzi, S., Tsitoura, C., Cossell, L., Strom, M. and Margrie, T. (2021) “A deep learning algorithm for 3D cell detection in whole mouse brain image datasets’ PLOS Computational Biology, 17(5), e1009074\nhttps://', 'title': 'This algorithm can also be used within the original\ncellfinder software for\nwhole-brain microscopy analysis', 'authors': [{'family-names': 'Tyson', 'given-names': 'A.'}, {'family-names': 'Rousseau', 'given-names': 'C.'}, {'family-names': 'Niedworok', 'given-names': 'C.'}, {'family-names': 'Keshavarzi', 'given-names': 'S.'}, {'family-names': 'Tsitoura', 'given-names': 'C.'}, {'family-names': 'Cossell', 'given-names': 'L.'}, {'family-names': 'Strom', 'given-names': 'M.'}, {'family-names': 'Margrie', 'given-names': 'T.'}]}}
```

How do I generate the CITATION.cff file from this?

**Comment by SimaoBolota-MetaCell** [comment]
Hi @neuromusic 

Thank you for the suggested changes. 

Regarding the CITATION.CFF it is created on the local plugin folder, so that it can be pushed directly with everything else from the plugin. 
I executed it locally now and it is working for me. 
Is the .cff not being created at all?

I will also add a console log of 'CITATION.CFF created at ../path' so that it can be easily identified.

Execution example: 

https://user-images.githubusercontent.com/99416933/200269561-f0ea524e-b739-42ab-9fbd-bb6ee0a99469.mov

**Comment by SimaoBolota-MetaCell** [comment]
Added console log of the cff file location.
Example:


https://user-images.githubusercontent.com/99416933/200292325-bbb3aea2-522e-4b30-abcf-3b23934d9687.mov

**Review by neuromusic — APPROVED** [review]

**Pull request merged**